src/Workspaces/Core/Portable/Workspace/Host/SyntaxTreeFactory/AbstractSyntaxTreeFactoryService.AbstractRecoverableSyntaxRoot.cs

```diff
@@ -21,7 +21,7 @@ internal struct SyntaxTreeInfo
 
             public SyntaxTreeInfo(string filePath, ParseOptions options, ValueSource<TextAndVersion> textSource, Encoding encoding, int length)
             {
-                FilePath = filePath;
+                FilePath = filePath ?? string.Empty;
```

Comment:
I have no idea how many things in our own code base, and our customer code bases would have to be hardened to support null here. There are enough places that *prevented* that before that allowing it now is likely to break things.